server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -102,7 +109,52 @@ private static XContentBuilder mappings() {
         }
     }
 
-    public void putSynonymsSet(String resourceName, SynonymsSet synonymsset, ActionListener<PutSynonymsAction.Response> listener) {
+    public void getSynonymsSet(String resourceName, int from, int size, ActionListener<SynonymsSetResult> listener) {
+        client.prepareSearch(SYNONYMS_ALIAS_NAME)
+            .setQuery(QueryBuilders.termQuery(SYNONYMS_SET_FIELD, resourceName))
```

Comment:
Should we also set up `.setPreference("_local")`?

Reply:
Adding those, I think they make the intention more clear. Thanks!